### PR TITLE
Improvements in Inverse Kinematics 

### PR DESCRIPTION
**Pull request by traversaro**
https://github.com/robotology/idyntree/pull/273/commits/51cc0dafc669934946d31af166a4362c7337872a 	[inverse-kinematics] Add support for weighting tasks separately
https://github.com/robotology/idyntree/pull/273/commits/aef9ee9db824f027233fea198821526b60eb7147   	[inverse-kinematics] Expose optimization parameters

### CODE REVIEW

File: src/inverse-kinematics/include/iDynTree/InverseKinematics.h [hunk 1]
```diff
@@ -166,6 +167,84 @@ class iDynTree::InverseKinematics
...
+
+    /**
+     * Sets Maximum Iteration.
+     *
+     * The default value for this parameter is 3000 .
+     *
+     * @param max_iter exits if iter>=max_iter (max_iter<0
+     *                   disables this check).
+     */
+    void setMaxIter(const int max_iter);
```
Comment (francesco-romano) [inline src/inverse-kinematics/include/iDynTree/InverseKinematics.h]: For the options I would have liked a sort of single method with key-value pairs, but I think that given the time constraints we can live it in this way

File: src/inverse-kinematics/include/private/InverseKinematicsData.h [hunk 2]
```diff
@@ -42,16 +42,16 @@ class internal::kinematics::InverseKinematicsData {
     InverseKinematicsData(const InverseKinematicsData&);
     InverseKinematicsData& operator=(const InverseKinematicsData&);
 
-    //!!!: I have to divide variables between the optimized one (buffers inside the Solver, except results and I/O variables here)
-    // and the "model" variables.
+    //!!!: The variables are divided among the optimized one (buffers inside the Solver, except results and I/O variables here),
```
Comment (francesco-romano) [inline src/inverse-kinematics/include/private/InverseKinematicsData.h]: If this is ok, you can remove the `!!!:` which is an Xcode tag (sort of todo)

File: src/inverse-kinematics/include/iDynTree/InverseKinematics.h (hunk 1 above)
Comment (traversaro) [inline src/inverse-kinematics/include/iDynTree/InverseKinematics.h]: Actually my design goal were mostly the compatibility with iKinIpOpt , I copied the methods from them. We can also have the key-value stuff, especially if we don't recreate IpOpt Problem at each call to solve, so we can set/get directly the parameters on the IpOpt problem instance as iKinIpOpt does.

**Comment by traversaro** [comment]
cc @pattacini

File: src/inverse-kinematics/include/private/InverseKinematicsData.h (hunk 2 above)
Comment (traversaro) [inline src/inverse-kinematics/include/private/InverseKinematicsData.h]: Good to know, I will remove it.

**Comment by traversaro** [comment]
Merging as the tests are failing also in `devel`, so it is not a big problem. : )

**Pull request merged**